doc/amoc_throttle.md

```diff
@@ -0,0 +1,152 @@
+## Overview
+
+Amoc throttle is a module that allows limiting number of users' actions per given interval, no matter how many users there are in a test.
+It works in both local and distributed environment, allows for dynamic rate changes during a test and exposes metrics which show number of requests and executions.
+
+Amoc throttle allows to set execution `Rate` per `Interval` or limit number of parallel executions when `Interval` is set to `0`.
+Each `Rate` is identified with a `Name`.
+Rate limiting mechanism is based on allowing execution in response to a request only when it would not exceed the given `Rate`.
+Amoc throttle makes sure that given `Rate` per `Interval` is maintained on possibly equal level the whole time.
+In other words, it will not create bursts of executions which would also technically result in the same rate in a given interval.
+Because of that, it may happen that the actual `Rate` will be little lower then demanded.
+It will however never be exceeded. 
+
+## API
+
+Functions provided by Amoc throttle:
+
+- `start(Name, Rate)`, `start(Name, Rate, Interval)`, `start(Name, Rate, Interval, NoOfProcesses)` - functions starting throttle mechanism for a given `Name` with a given `Rate`.
+They can also accept `Interval` (default is one minute) and ` NoOfProcesses` (default is 10).
+`Name` is needed to identify the rate as one test can have different rates for different tasks.
+`Interval` is given in milliseconds and can be changed to a different value for convenience or higher granularity.
+It also accepts a special value of `0` which limits the number of parallel executions associated with `Name` to `Rate`.
+- `send(Name, Msg)`, `send(Name, Pid, Msg)` - sends a given message `Msg` either to self or a given `Pid` when rate for `Name` allows for that.
+May be used to schedule tasks.
+- `send_and_wait(Name, Msg)` - sends and receives given message `Msg`.
+Can be used to halt execution if we want a process to be idle when waiting for rate increase or other processes finishing their tasks.
+- `run(Name, Fn)` - executes a given function `Fn` when it does not exceed rate for `Name`.
+`Fn` is executed in a context of a new process spawned on the same node on which process executing `run/2` runs,
+so call to `run/2` is non-blocking.
+This function is used internally by both `send` functions and `send_and_wait/2`,
+so all those actions will be limited to the same rate when called with the same `Name`.
+- `pause(Name)` - pauses executions for given `Name` as if `Rate` was set to `0`.
+Does not stop scheduled rate changes.
+- `resume(Name)` - resumes executions for given `Name`, so `Rate` and `Interval` are set to the values they are supposed to
```

Comment:
```suggestion
- `resume(Name)` - resumes executions for a given `Name`, so the `Rate` and `Interval` are set to the correct values
```

??

Reply:
- `resume(Name)` - resumes the executions for a given `Name`, so the `Rate` and `Interval` are set to the values (which may be different to the ones set before running the `pause/1` function). 
??